coins/coins.go

```diff
@@ -0,0 +1,252 @@
+// Package coins provides types, conversions and exchange calculations for dealing
+// with cryptocurrency coins and ECR20 tokens.
+package coins
+
+import (
+	"fmt"
+	"math/big"
+	"strconv"
+
+	"github.com/cockroachdb/apd/v3"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// PiconeroAmount represents some amount of piconero (the smallest denomination of monero)
+type PiconeroAmount apd.Decimal
```

Comment:
why does this need to be `apd.Decimal`? it'll always be a positive integer value, so uint64 or *big.Int would also be fine

Reply:
It doesn't have to be, but I think there is value in treating all the currencies the same way. I didn't want to take it on in this PR, but XMR, ETH and ERC20 tokens could all be handled with a single class that tracks the decimals, unit-name/currency-name, etc. if we want to do it that way in the future. It's convenient to go back and forth with XMR just by adjusting the exponent. After working with `apd.Decimal` for a while, I really like it. For money, it seems like a definite improvement over `big.Int`.